kafka-ui-api/src/main/java/com/provectus/kafka/ui/util/JmxPoolFactory.java

```diff
@@ -1,20 +1,31 @@
 package com.provectus.kafka.ui.util;
 
+import com.provectus.kafka.ui.model.JmxConnectionInfo;
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 import javax.management.remote.JMXConnector;
 import javax.management.remote.JMXConnectorFactory;
 import javax.management.remote.JMXServiceURL;
+import javax.rmi.ssl.SslRMIClientSocketFactory;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.pool2.BaseKeyedPooledObjectFactory;
 import org.apache.commons.pool2.PooledObject;
 import org.apache.commons.pool2.impl.DefaultPooledObject;
 
 @Slf4j
-public class JmxPoolFactory extends BaseKeyedPooledObjectFactory<String, JMXConnector> {
+public class JmxPoolFactory extends BaseKeyedPooledObjectFactory<JmxConnectionInfo, JMXConnector> {
 
   @Override
-  public JMXConnector create(String s) throws Exception {
-    return JMXConnectorFactory.connect(new JMXServiceURL(s));
+  public JMXConnector create(JmxConnectionInfo info) throws Exception {
+    Map<String, Object> env = new HashMap<>();
+    if (StringUtils.isNotEmpty(info.getUsername()) && StringUtils.isNotEmpty(info.getPassword())) {
+      env.put("jmx.remote.credentials", new String[]{info.getUsername(), info.getPassword()});
+    }
+    env.put("com.sun.jndi.rmi.factory.socket", new SslRMIClientSocketFactory());
```

Comment:
Will this work for none SSL connections?

Reply:
Damn, I once again forgot about these things being optional. 
Turns out it does not work with non-SSL connections, I pushed the changed to make this optional.